modules/transactions/src/main/java/org/apache/ignite/internal/tx/storage/state/rocksdb/TxStateRocksDbSharedStorage.java

```diff
@@ -0,0 +1,193 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.tx.storage.state.rocksdb;
+
+import static java.util.Collections.reverse;
+import static java.util.stream.Collectors.toList;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.IntSupplier;
+import org.apache.ignite.internal.close.ManuallyCloseable;
+import org.apache.ignite.internal.lang.IgniteInternalException;
+import org.apache.ignite.internal.rocksdb.flush.RocksDbFlusher;
+import org.apache.ignite.internal.tx.storage.state.TxStateTableStorage;
+import org.apache.ignite.internal.util.IgniteSpinBusyLock;
+import org.apache.ignite.internal.util.IgniteUtils;
+import org.rocksdb.ColumnFamilyDescriptor;
+import org.rocksdb.ColumnFamilyHandle;
+import org.rocksdb.ColumnFamilyOptions;
+import org.rocksdb.DBOptions;
+import org.rocksdb.Options;
+import org.rocksdb.ReadOptions;
+import org.rocksdb.RocksDB;
+import org.rocksdb.WriteOptions;
+
+/**
+ * RocksDb implementation of {@link TxStateTableStorage}.
+ */
+public class TxStateRocksDbSharedStorage implements ManuallyCloseable {
+    static {
+        RocksDB.loadLibrary();
+    }
+
+    /** Column family name for transaction states. */
+    private static final String TX_STATE_CF = new String(RocksDB.DEFAULT_COLUMN_FAMILY, StandardCharsets.UTF_8);
+
+    /** Rocks DB instance. */
+    private volatile RocksDB db;
+
+    /** RocksDb database options. */
+    private volatile DBOptions dbOptions;
+
+    /** Write options. */
+    final WriteOptions writeOptions = new WriteOptions().setDisableWAL(true);
+
+    /** Read options for regular reads. */
+    final ReadOptions readOptions = new ReadOptions();
+
+    /** Database path. */
+    private final Path dbPath;
+
+    /** RocksDB flusher instance. */
+    private volatile RocksDbFlusher flusher;
+
+    /** Prevents double stopping the storage. */
+    private final AtomicBoolean stopGuard = new AtomicBoolean();
+
+    /** Busy lock to stop synchronously. */
+    private final IgniteSpinBusyLock busyLock = new IgniteSpinBusyLock();
+
+    private final ScheduledExecutorService scheduledExecutor;
+
+    /** Thread pool to execute after-flush actions. */
+    private final ExecutorService threadPool;
+
+    /** Supplier for the value of delay for scheduled database flush. */
+    private final IntSupplier flushDelaySupplier;
+
+    /**
+     * Constructor.
+     *
+     * @param dbPath Database path.
+     * @param scheduledExecutor Scheduled executor.
+     */
+    public TxStateRocksDbSharedStorage(
```

Comment:
Not all parameters are listed in the documentation.
And also what error there might be and why it might occur.